flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -34,7 +34,7 @@
 /**
  * Context object with optional arguments for a Flink Scan.
  */
-class ScanContext implements Serializable {
+public class ScanContext implements Serializable {
```

Comment:
I think we should just have the minimal set of changes to this file, rather than trying to fix the duplication between this and `FlinkSource.Builder`. Also, we should probably add `@Internal` because this is becoming public.

Reply:
To clarify, the FLIP-27 source change wasn't trying to fix the duplication btw `ScanContext` and current `FlinkSource.Builder`. Existing code stays unchanged. I was just trying to avoid the duplication with the new FLIP-27 `IcebergSource.Builder` by making `ScanContext` user-facing public and letting users directly construct the `ScanContext`. 

With that said, it seems that both @openinx  and @rdblue  prefer the current `FlinkSource.Builder` model where `ScanContext` APIs were duplicated in the source builder. I will make the change to follow the current pattern.